src/utils/viewer/viewer.cpp

```diff
@@ -728,7 +747,7 @@ void Viewer::drawEvent() {
   Mn::GL::defaultFramebuffer.bind();
 
   imgui_.newFrame();
-
+  profiler_.beginFrame();
```

Comment:
Not sure if this was intentional, but what gets submitted to GL between `beginFrame()` / `endFrame()` adds up to the per-frame CPU and GPU duration in the output. As the code is now, only ImGui drawing is measured, which I'd say is the inverse of what we want :) 

I'd say everything from the beginning of `drawEvent()` until after `blitRgbaToDefault()` (or maybe without the clears, without the blits, without the physics?) is of interest, the rest not really. Feel free to experiment and see which measurement scope gives the most useful results for datasets of various complexity.

(I have to admit the frame profiler is a bit lacking in the sense that it doesn't allow you to take multiple measurement points in a frame, only two. It could be worked around by having multiple profiler instances in a frame, for example dedicate one to measure just the cost of physics calculation by wrapping its begin/end around the physics step block. Just an idea, feel free to ignore that, you have a ton of other things to do and I don't want you to get stuck on this for weeks :))

Reply:
Thanks, that's fixed!

Reply:
@mosra:
Thank you for helping @vauduong . Sincerely appreciate your high quality review work!
I did not fully read the wiki about the profiler, can it be used as a stop-watch meaning it has functions such as `pause()` and `resume()` so that user can skip a piece of code in between (`beginFrame()` and `endFrame()`) when measuring the performance?